[docs] add another kube-rbac-proxy example (#4280)
deckhouse/deckhouse: modules/300-prometheus/docs/FAQ.md

To enable secure access to metrics, we strongly recommend using **kube-rbac-proxy**.

### An example of collecting metrics securely from an application inside a cluster

> To configure application metrics protection using `kube-rbac-proxy` and then build metrics from it using Prometheus tools, perform the following steps:

Reviewer: ```suggestion
Do the following to set up application metrics protection via the `kube-rbac-proxy` with the subsequent metrics scraping using Prometheus tools:
```

1. Create a new `ServiceAccount` with the following permissions:

   ```yaml
   ---
   apiVersion: v1
   kind: ServiceAccount
   metadata:
     name: rbac-proxy-test
   ---
   apiVersion: rbac.authorization.k8s.io/v1
   kind: ClusterRoleBinding
   metadata:
     name: rbac-proxy-test
   roleRef:
     apiGroup: rbac.authorization.k8s.io
     kind: ClusterRole
     name: d8:rbac-proxy
   subjects:
   - kind: ServiceAccount
     name: rbac-proxy-test
     namespace: default
   ```

   > The example uses the `d8:rbac-proxy` built-in Deckhouse `ClusterRole`.

> 2. Create a configuration for `kube-rbac-proxy`:

Reviewer: ```suggestion
2. Create a configuration for the `kube-rbac-proxy`:
```

   ```yaml
   ---
   apiVersion: v1
   kind: ConfigMap
   metadata:
     name: rbac-proxy-config-test
     namespace: rbac-proxy-test
   data:
     config-file.yaml: |+
       authorization:
         resourceAttributes:
           namespace: default
           apiVersion: v1
           resource: services
           subresource: proxy
           name: rbac-proxy-test
   ```

   > Get more information on authorization attributes in the [Kubernetes documentation](https://kubernetes.io/docs/reference/access-authn-authz/authorization).

> 3. Create `Service` and `Deployment` for your application, where `kube-rbac-proxy` takes the position of a sidecar container:

Reviewer: ```suggestion
3. Create `Service` and `Deployment` for your application with the `kube-rbac-proxy` as a sidecar container:
```

       app: rbac-proxy-test
   ---
   apiVersion: apps/v1
   kind: Deployment
   metadata:
     name: rbac-proxy-test
   spec:
     replicas: 1
     selector:
       matchLabels:
         app: rbac-proxy-test
     template:
       metadata:
         labels:
           app: rbac-proxy-test
       spec:
         securityContext:
           runAsUser: 65532
         serviceAccountName: rbac-proxy-test
         containers:
         - name: kube-rbac-proxy
           image: quay.io/brancz/kube-rbac-proxy:v0.14.0
           args:
           - "--secure-listen-address=0.0.0.0:8443"
           - "--upstream=http://127.0.0.1:8081/"
           - "--config-file=/kube-rbac-proxy/config-file.yaml"
           - "--logtostderr=true"
           - "--v=10"
           ports:
           - containerPort: 8443
             name: https-metrics
           volumeMounts:
           - name: config
             mountPath: /kube-rbac-proxy
         - name: prometheus-example-app
           image: quay.io/brancz/prometheus-example-app:v0.1.0
           args:
           - "--bind=127.0.0.1:8081"
         volumes:
         - name: config
           configMap:
             name: rbac-proxy-config-test
   ```

> 4. Assign the necessary rights to the resource to Prometheus:

Reviewer: ```suggestion
4. Add the necessary resource permissions to Prometheus:
```

   ```yaml
   ---
   apiVersion: rbac.authorization.k8s.io/v1
   kind: ClusterRole
   metadata:
     name: rbac-proxy-test-client
   rules:
   - apiGroups: [""]
     resources: ["services/proxy"]
     verbs: ["get"]
   ---
   apiVersion: rbac.authorization.k8s.io/v1
   kind: ClusterRoleBinding
   metadata:
     name: rbac-proxy-test-client
   roleRef:
     apiGroup: rbac.authorization.k8s.io
     kind: ClusterRole
     name: rbac-proxy-test-client
   subjects:
   - kind: ServiceAccount
     name: prometheus
     namespace: d8-monitoring
   ```

> After step 4, your application's metrics should appear in Prometheus.

Reviewer: ```suggestion
After step 4, your application's metrics should become available in Prometheus.
```
